packages/react/src/migrations/update-19-6-0/update-ssr-server-port.spec.ts

```diff
@@ -0,0 +1,202 @@
+             * DO NOT REMOVE IF USING @nx/react:module-federation-dev-ssr executor
+             * to serve your Host application with this Remote application.
+             * This message allows Nx to determine when the Remote is ready to be
+             * consumed by the Host.
+             */
+            process.send?.('nx.server.ready');
+        });
+        server.on('error', console.error);
+        "
+      `);
+    } else {
+      fail('The file product/server.ts does not exist in the tree');
+    }
+
+    if (tree.exists('shell/server.ts')) {
+      tree.write(
+        'shell/server.ts',
+        tree
+          .read('shell/server.ts', 'utf-8')
+          .replace(
+            'const port = 4200;',
+            `const port = process.env['PORT'] || 4200;`
+          )
+      );
+
+      updateSsrServerPort(tree);
+      expect(tree.read('shell/server.ts', 'utf-8')).toContain(
+        `port = ${shellPort}`
+      );
+      expect(tree.read('shell/server.ts', 'utf-8')).toMatchInlineSnapshot(`
+        "import * as path from 'path';
+        import express from 'express';
+        import cors from 'cors';
+        import { handleRequest } from './src/main.server';
+        const port = 4200;
+        const app = express();
+        const browserDist = path.join(process.cwd(), 'dist/shell/browser');
+        const indexPath = path.join(browserDist, 'index.html');
+        app.use(cors());
+        app.get('*.*', express.static(browserDist, {
+            maxAge: '1y',
+        }));
+        app.use('*', handleRequest(indexPath));
+        const server = app.listen(port, () => {
+            console.log(\`Express server listening on http://localhost:\${port}\`);
+        });
+        server.on('error', console.error);
+        "
+      `);
+    } else {
+      fail('The file shell/server.ts does not exist in the tree');
+    }
+  });
+
+  it('should update a host project server file', async () => {
+    await hostGenerator(tree, {
+      name: 'host',
+      e2eTestRunner: 'none',
+      unitTestRunner: 'none',
+      ssr: true,
+      linter: Linter.EsLint,
+      projectNameAndRootFormat: 'as-provided',
+      style: 'css',
+    });
+
+    const hostPort = readProjectConfiguration(tree, 'host').targets.serve
+      .options.port;
+
+    if (tree.exists('host/server.ts')) {
+      tree.write(
+        'host/server.ts',
+        tree
+          .read('host/server.ts', 'utf-8')
+          .replace(
+            'const port = 4200;',
+            `const port = process.env['PORT'] || 4200;`
+          )
+      );
+
+      updateSsrServerPort(tree);
+
+      expect(tree.read('host/server.ts', 'utf-8')).toContain(
+        `port = ${hostPort}`
+      );
+      expect(tree.read('host/server.ts', 'utf-8')).toMatchInlineSnapshot(`
+        "import * as path from 'path';
+        import express from 'express';
+        import cors from 'cors';
+        import { handleRequest } from './src/main.server';
+        const port = 4200;
```

Comment:
I think we should keep `process.env['PORT']` in here, but just update the default port if the env var is undefined.